Process Emergency Stop

sithlord48 opened the pull request:
Emergency Stop is now no the testGUI and when pressed it will either send M112 directly or if the printer is in a busy state it will set the printers state to "STOP" and atcore will do the rest.

sithlord48 committed:
Process Stop Emergency Stop Command

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>

sithlord48 committed:
Add Emergency Stop Button to ui and connect it in mainwindow

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>

patrickelectric commented on testclient/mainwindow.cpp:
```diff
@@ -263,6 +264,15 @@ void MainWindow::printPBClicked()
+        core->setState(STOP);
+        ui->printPB->setText(tr("Print File"));
+    } else {
+        core->serial()->pushCommand(QString("M112").toLocal8Bit());
```
Here, the command need to be sent to atcore.

patrickelectric replied:
use gcode namespace

sithlord48 replied:
:heavy_check_mark:

patrickelectric commented on testclient/mainwindow.cpp:
```diff
@@ -263,6 +264,15 @@ void MainWindow::printPBClicked()
 
+void MainWindow::stopPBClicked()
+{
+    if (core->state() == BUSY) {
```
It's better to send only the stop command to the printer. Don't have problem if you send more than one.

patrickelectric commented on src/atcore.cpp:
```diff
@@ -172,6 +172,10 @@ void AtCore::print(const QString &fileName)
         else if (printerState == STOP) {
             qDebug() << tr("Stop State");
+            //slip stop command into the gcodestream. Let the firmware do the rest.
+            QString stopString("M112");
```
use gcode namespace

sithlord48 replied:
:heavy_check_mark:

patrickelectric reviewed (changes requested):
Use setState and state functions

patrickelectric committed:
atcore: Change functions to use state() and setState()

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
atcore: Change ifelse to switch case

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric reviewed (changes requested):
Change from if else to switch case

patrickelectric committed:
atcore: Change from d->serial to serial()

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric reviewed (changes requested):
Change from d->serial to serial()

patrickelectric committed:
atcore,plugins: Add translate function

The plugin need to translate the generic Gcode to the firmware language,
the translate function will perform that for different kinds of communnications
protocols.

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
atcore: Change d->fwPlugin to plugin()

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
atcore: Add setSerial and setPlugin

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
atcore: Change char to QChar

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

sithlord48 committed:
Merge pull request #2 from patrickelectric/sith_rev

atcore: change functions to use state() and setState()

sithlord48 committed:
Have serial device send fw request. atcore can't translate with out a
plugin loaded after all

Signed-off-by: Chris Rizzitello <sithlord48@gmail.com>
